### ISSUE TITLE
flags-import matches partial arguments

### ISSUE THREAD

**Issue by AndrewPaulChester** [guildai/guildai#31]
When importing guild flags using the flags-import directive in the guild.yml file, any arguments defined by argparse which are substrings of the specified flags are also imported. See minimal working example below:

```
import argparse

parser = argparse.ArgumentParser(description="RL")
parser.add_argument("--import", default="test_file", help="import file")
parser.add_argument("--port", default="sydney", help="port of origin")
parser.add_argument("--not-imported", default="a2c", help="for demonstration")

parser.parse_args()
```

```
- model: argstest
  operations:
    train:
      main: guild_flag_test
      flags-import: 
        -import
```

running `guild run argstest:train`
gives the following output (port flag unexpected): 
```
You are about to run argstest:train
  import: test_file
  port: sydney
Continue? (Y/n) n
```

**Comment by gar1t** [comment]
You want to spell your `-import` with a space after the `-`:

``` yaml
      flags-import: 
        - import
```

That space is required to tell YAML that it's a list item. Otherwise it's treated as a string.

I'm looking into why Guild is doing that weird pattern match with `-import` but regardless, that's the fix for your case.

Let's keep this open until the mysterious pattern match behavior is fixed though.

**Comment by gar1t** [comment]
Okay, this is all cleaned up in `0.6.5` the pre-release is available for test otherwise the official release goes out later this week.

The behavior now will catch the original spec (i.e. where the item is `-import` rather than `- import`) as a Guild file error and prints a message saying what it's expecting.

Thanks for the carefully written report!

**Comment by gar1t** [comment]
I created a script to confirm that this behavior is addressed in 0.6.5.

https://github.com/guildai/issues/tree/master/issue-31

Closing. Thanks again!
